### PR TITLE
Introduction to Node.Js (First version)

### PR DESCRIPTION
**Pull request by avishekjana**

### CODE REVIEW

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Text: Introduction to Node.Js
+
+Hello and welcome to WD201! 
+In this course we will learn about a JavaScript framework called Node.Js. I'm assuming that you've already completed the [Web Development 101 course](https://www.pupilfirst.school/courses/1087/curriculum). This means that, you're quite familiar with JavaScript, right?
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:4]:
```suggestion
In this course, we will learn about a JavaScript framework called Node.js. I'm assuming that you've already completed the [Web Development 101 course](https://www.pupilfirst.school/courses/1087/curriculum). This means that you're quite familiar with JavaScript, right?
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+To recap, JavaScript is a **Programming Language**,  that runs on end user's browser, and allows us to calculate, manipulate and validate data. Where HTML and CSS are languages that give structure and style to web pages, JavaScript improves the user experience of the web page by converting it from a static page into an interactive one.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:6]:
```suggestion
To recap, JavaScript is a **Programming Language**,  that runs on the end user's browser, and allows us to calculate, manipulate and validate data. Where HTML and CSS are languages that give structure and style to web pages, JavaScript improves the user experience of the web page by converting it from a static page into an interactive one.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+Now, have you ever wondered, the static HTML webpages or the forms that you've designed, how to make them dynamic? How to store that form's data in a database? Or what if, you want to send an confirmation email to an user who submits the form?
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:8]:
```suggestion
Now, have you ever wondered, about the static HTML webpages or the forms that you've designed, and how to make them dynamic? How to store that form's data in a database? Or, what if, you want to send a confirmation email to a user who submits the form? 
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+In this course we will learn about a JavaScript framework called Node.Js. I'm assuming that you've already completed the [Web Development 101 course](https://www.pupilfirst.school/courses/1087/curriculum). This means that, you're quite familiar with JavaScript, right?
+
+To recap, JavaScript is a **Programming Language**,  that runs on end user's browser, and allows us to calculate, manipulate and validate data. Where HTML and CSS are languages that give structure and style to web pages, JavaScript improves the user experience of the web page by converting it from a static page into an interactive one.
+
+Now, have you ever wondered, the static HTML webpages or the forms that you've designed, how to make them dynamic? How to store that form's data in a database? Or what if, you want to send an confirmation email to an user who submits the form? 
+This is where the role of **server-side programming** comes in.
+
+## What is server-side programming?
+Web browsers communicate with web servers using the HyperText Transfer Protocol (HTTP). When you click a link on a web page or submit a form, an HTTP request is sent from your browser to the target web server.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:12]:
```suggestion
Web browsers communicate with web servers using the Hypertext Transfer Protocol (HTTP). When you click a link on a web page or submit a form, an HTTP request is sent from your browser to the target web server.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+On the other hand, web servers wait for client request messages. It processes them when they arrive, and responds back to the browser with an HTTP response message.
+
+The body of a successful response contains the requested resource (for example a new HTML page, or an image, etc...), which could then be displayed by the browser.
+
+### Dynamic websites
+A dynamic website is one where some response content is generated dynamically, only when needed. On a dynamic website, the HTML pages are normally created by fetching data from a database, then inserting that data into certain placeholders in HTML template.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:18]:
```suggestion
A dynamic website is one where some response content is generated dynamically, only when needed. On a dynamic website, the HTML pages are normally created by fetching data from a database, and then inserting that data into certain placeholders in HTML template. 
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+On the other hand, web servers wait for client request messages. It processes them when they arrive, and responds back to the browser with an HTTP response message.
+
+The body of a successful response contains the requested resource (for example a new HTML page, or an image, etc...), which could then be displayed by the browser.
+
+### Dynamic websites
+A dynamic website is one where some response content is generated dynamically, only when needed. On a dynamic website, the HTML pages are normally created by fetching data from a database, then inserting that data into certain placeholders in HTML template. 
+
+Most of the **code** to support a dynamic website must run on the server. Creating this code is known as **server-side programming** or **back-end programming**.
+
+Now imagine, what if you could use your JavaScript knowledge to write back-end or server-side logic? This is where the role of Node.Js comes in.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:22]:
```suggestion
Now imagine, what if you could use your JavaScript knowledge to write back-end or server-side logic? This is where the role of Node.js comes in.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+## What is Node.Js?
+Node.js is a framework for writing server-side JavaScript applications. Node.js is open-source and completely free, it's used by thousands of developers around the world.
+Node is often used to build back end services that communicate with client-side applications.
+
+Every browser has their own JavaScript engine that converts JavaScript into code that a computer can understand. For example, Microsoft Edge uses Chakra, Firefox uses spidermonkey, and chrome uses V8.
+
+Before Node.JS, you could only run JavaScript in the browser. In 2009, Ryan Dahl came up with the idea of executing JavaScript code on the server-side, by using Chrome's v8 engine. Node.Js is a platform which is actually built on top of **Chrome's JavaScript Runtime environment**. Which makes it blazing fast when it comes down to the performance.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:30]:
```suggestion
Before Node.js, you could only run JavaScript in the browser. In 2009, Ryan Dahl came up with the idea of executing JavaScript code on the server-side, by using Chrome's v8 engine. Node.js is a platform which is actually built on top of **Chrome's JavaScript Runtime environment**. Which makes it blazing fast when it comes down to the performance.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+Most of the **code** to support a dynamic website must run on the server. Creating this code is known as **server-side programming** or **back-end programming**.
+
+Now imagine, what if you could use your JavaScript knowledge to write back-end or server-side logic? This is where the role of Node.Js comes in.
+
+## What is Node.Js?
+Node.js is a framework for writing server-side JavaScript applications. Node.js is open-source and completely free, it's used by thousands of developers around the world.
+Node is often used to build back end services that communicate with client-side applications.
+
+Every browser has their own JavaScript engine that converts JavaScript into code that a computer can understand. For example, Microsoft Edge uses Chakra, Firefox uses spidermonkey, and chrome uses V8.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:28]:
```suggestion
Every browser has their own JavaScript engine that converts JavaScript into code that a computer can understand. For example, Microsoft Edge uses Chakra, Firefox uses SpiderMonkey, and chrome uses V8.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+Before Node.JS, you could only run JavaScript in the browser. In 2009, Ryan Dahl came up with the idea of executing JavaScript code on the server-side, by using Chrome's v8 engine. Node.Js is a platform which is actually built on top of **Chrome's JavaScript Runtime environment**. Which makes it blazing fast when it comes down to the performance.
+
+So, let's get started with the installation.
+
+# Script: Installation of Node.Js
+
+In this video, I'll show you, how easily you can setup your computer to write Node.Js applications.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:36]:
```suggestion
In this video, I'll show you, how easily you can setup your computer to write Node.js applications.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+I'm assuming that you're either running macOS, or some variant of Linux, or you're probably using WSL (Ubuntu) inside of Windows.
+
+## Install Node.Js using NVM
+
+There are a number of different ways you can install Node.Js on your system. However, one of the most popular is *nvm.* NVM stands for Node.Js Version Manager. It allows us to quickly install and use different versions of node in our computer.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:41]:
```suggestion
There are a number of different ways you can install Node.js on your system. However, one of the most popular is *nvm.* NVM stands for Node Version Manager. It allows us to quickly install and use different versions of Node.js in our computer.
```

File: introduction-to-nodejs/what-is-node-js-with-installation/README.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+*EXPLAIN Output*
+
+To use a specific version in a project, run:
+
+`nvm use 17`
+
+### 6. Check npm version
+
+**npm** is a package manager for the JavaScript programming language. It consists of a command line client, also called **npm**, and an online database of public and private packages. Open source developers use npm to share and borrow packages.
```
Comment (vikispeaks) [inline introduction-to-nodejs/what-is-node-js-with-installation/README.md:132]:
```suggestion
**npm** is a package manager for the JavaScript programming language. It consists of a command-line client, also called **npm**, and an online database of public and private packages. Open-source developers use npm to share and borrow packages.
```

**Review by vikispeaks — CHANGES_REQUESTED** [review]
Please update all references to Node.js right.

**Comment by avishekjana** [comment]
@vikispeaks, I've made all suggested changes, please approve and close the PR's

**Pull request merged**